src/simulation.jl

```diff
@@ -1,38 +1,47 @@
-using Base.Iterators
 using CombinatorialSpaces
 using ComponentArrays
 using LinearAlgebra
 using MLStyle
 using PreallocationTools
 
-const gensim_in_place_stub = Symbol("GenSim-M")
+const GENSIM_INPLACE_STUB = Symbol("GenSim-M")
+const NO_STUB_RETURN = Symbol("NOSTUB")
 
 abstract type GenerationTarget end
 
-struct CPUTarget <: GenerationTarget end
-struct CUDATarget <: GenerationTarget end
+abstract type CPUBackend <: GenerationTarget end
+abstract type CUDABackend <: GenerationTarget end
 
+struct CPUTarget <: CPUBackend end
+struct CUDATarget <: CUDABackend end
+
+# TODO: Add exceptions for invalid calls
```

Comment:
This TODO was more so that if what we pass invalid data into some `AbstractCall`, then instead of falling into an `else` condition, which is based on an assumption, we instead error out. I'll clarify this comment.